caffe2/core/blob.h

```diff
@@ -241,16 +233,16 @@ class CAFFE2_API Blob final : public c10::intrusive_ptr_target {
   void Deserialize(const BlobProto& proto);
 
  private:
-  /**
-   * @brief A destroy call that is used to properly deconstruct objects.
-   */
-  template <class T>
-  static void Destroy(void* pointer) {
-    delete static_cast<T*>(pointer);
+  void free_() {
+   if (has_ownership_) {
+     CAFFE_ENFORCE(pointer_ != nullptr, "Can't have ownership of nullptr");
+     (*meta_.dtor())(pointer_, 1);
```

Comment:
wait a second - before the delete call was actually freeing the memory (calling operator `delete`), now it just calls placement destructor and leaves memory hanging around

Reply:
fixed